ruff_lsp/settings.py

```diff
@@ -94,8 +105,8 @@ def lint_args(settings: UserSettings) -> list[str]:
 def lint_run(settings: UserSettings) -> Run:
     """Get the `lint.run` setting from the user settings."""
     if "lint" in settings and "run" in settings["lint"]:
-        return settings["lint"]["run"]
+        return Run(settings["lint"]["run"])
```

Comment:
How is this change related to Jupyter notebook support? What motivates the change?

Reply:
With a new document support, the values are being used in multiple places. Earlier, a string literal was used directly but that means the value needs to be repeated in multiple places. The values being `"onType"`, `"onSave"`. The downsides of using a string literal is we've no way of knowing that a small type like `"ontype"` ("t" is lowercase) will break the code without running it. With an enum, the editor can provide auto-completion, diagnostics, and rename wherever they're being utilized.